### PR TITLE
SOA parameters for pdns 4.4

### LINKED ISSUE #31: PowerDNS 4.4.0 and beyond
Starting PowerDNS 4.4.0, settings for default SOA values were removed:
https://doc.powerdns.com/authoritative/upgrading.html#removed-features

Now, handle_soa must return additional values: refresh, retry, expire and ttl

### PR DESCRIPTION
**Pull request by eburcat**
Closes #31

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,6 +30,14 @@ environment variables override those:
 
 `NIPIO_SOA_NS`: SOA name server.
 
+`NIPIO_SOA_REFRESH`: SOA refresh.
+
+`NIPIO_SOA_RETRY`: SOA retry.
+
+`NIPIO_SOA_EXPIRY`: SOA expiry.
+
+`NIPIO_SOA_MINIMUM`: SOA minimum ttl.
```
Comment (vanillajonathan) [inline README.md:39]: Uppercase TTL since it is an acronym. Expand TTL into "SOA minimum time-to-live (TTL)".

Comment (vanillajonathan) [reply README.md:39]: Perhaps `NIPIO_SOA_MINIMUM_TTL` would describe it better?

File: nipio/backend.conf [hunk 2]
```diff
@@ -23,6 +23,14 @@ id=1
 hostmaster=hostmaster@nip.io.example
 # Name server
 ns=ns1.nip.io.example
+# Refresh
```
Comment (vanillajonathan) [inline nipio/backend.conf:26]: Maybe specify that the unit is seconds.
"# Refresh (in seconds, 3600 is 60 minutes)"

**Comment by vanillajonathan** [comment]
The [Dockerfile](https://github.com/exentriquesolutions/nip.io/blob/master/Dockerfile) needs to be updated to target a newer version of Alpine Linux that has pdns version 4.4 or later.

@rses Take a look at this PR!

**Comment by rses** [comment]
Maybe I'm going blind, I can't see changes to Dockerfile ? Does alpine 3.7 install pds 4.4 ?

**Comment by vanillajonathan** [comment]
@rses, there are no changes made to the Dockerfile, that needs to be done.

https://pkgs.alpinelinux.org/packages?name=pdns&branch=v3.7
https://pkgs.alpinelinux.org/packages?name=pdns&branch=v3.15

Update to v3.14 or later, such as v3.15.

**Comment by vanillajonathan** [comment]
@eburcat & @rses it seems the tests fail.

**Comment by vanillajonathan** [comment]
@eburcat & @rses anyone of you know anything about this?

Errors:

test_configure_with_environment_variables_set 

    AssertionError: Expected
    <ns1.example.com hostmaster@example.com 40 41 42 43 99> to be equal to
    <ns1.example.com hostmaster@example.com 99 40 41 42 43>, but was not.

test_configure_with_full_config

    AssertionError: Expected
    <ns1.nip.io.test emailaddress@nip.io.test 56 57 58 59 55> to be equal
    <ns1.nip.io.test emailaddress@nip.io.test 55 56 57 58 59>, but was not.

**Comment by vanillajonathan** [comment]
It seems the order was wrong, I fixed that so now the unit tests work. They pass on @eburcat repo https://github.com/eburcat/nip.io/runs/7803372761

But GitHub Actions is acting weird and the CI fails on this repo with a "Resource not accessible by integration" error which seems to be a issue others experienced too: https://github.com/actions/first-interaction/issues/10

**Review by Coins2payroll — APPROVED** [review]

**Review by beluc — APPROVED** [review]

**Pull request merged**